scripts/Engineering/EnginX.py

```diff
@@ -0,0 +1,603 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import (absolute_import, division, print_function)
+import mantid.simpleapi as simple
+import Engineering.EnggUtils as Utils
+import os
+import csv
+from platform import system
+from shutil import copy2
+
+
+def main(vanadium_run, user, focus_run, **kwargs):
+    """
+    sets up all defaults, then calls run
+
+
+    @param vanadium_run :: the run number of the vanadium to be used
+    @param user :: the username to save under
+    @param focus_run :: the run number to focus
+    Kwargs:
+        force_vanadium (bool): forces creation of vanadium even if one for the run already exists
+        pre_process_run (bool): set whether or not to pre-process run before focusing
+        crop_type (string): what method of cropping to use
+        ceria_run (string): the run number of the ceria to use
+        crop_on (string): the bank of spectrum to crop on if cropping
+        grouping_file (string): the path of the grouping file for texture focusing
+        directory (string): the path of the directory to save to
+
+    """
+    # Set all values at top,
+    # get whether or not to force a vanadium, default value is false
+    do_van = kwargs.get("force_vanadium", False)
+    # get whether or not to pre-process the run before focusing, default is false
+    do_pre_process = kwargs.get("pre_process_run", False)
+    # get whether or not to force re-create the calibration, default is false
+    do_cal = kwargs.get("force_cal", False)
+    cropped = kwargs.get("crop_type", None)
+    ceria_run = kwargs.get("ceria_run", "241391")
+    crop_name = kwargs.get("crop_name", "cropped")
+    crop_on = kwargs.get("crop_on", None)
+    params = kwargs.get("params", None)
+    time_period = kwargs.get("time_period", None)
+    grouping_file = kwargs.get("grouping_file", None)
+    if crop_on is None and cropped is not None:
+        print("You must enter a value to crop_on if you enter a cropping method")
+        return
+    if system() == "Windows":
+        path = os.path.join("/", "EnginX_Mantid")
+    else:
+        path = os.path.expanduser("~/EnginX_Mantid")
+    directory = kwargs.get("directory", path)
```

Comment:
Can you group all the `kwargs` properties together and separate the paths by some whitespace, just to make it more clear what the use can be expected to change